### ISSUE TITLE
effect does not merge environments on sequenceS

### ISSUE THREAD

**Issue by mohaalak** [mikearnaldi/matechs-effect#112]
let's say we have this code

```typescript
import * as EFF from '@matechs/effect/lib/effect';
import {sequenceS} from 'fp-ts/lib/Apply'

const seqSeffect = sequenceS(EFF.effect)

const firstEffect = () => EFF.accessM(({a}: {a: number}) => EFF.pure(a))
const secondEffect = () => EFF.accessM(({b}: {b: number}) => EFF.pure(b) )

seqSeffect({first: firstEffect(), second: secondEffect()})
```

this will raise this error 

```
Type 'Effect<{ b: number; }, never, number>' is not assignable to type 'Effect<{ a: number; }, never, any>'.
  Property 'a' is missing in type '{ b: number; }' but required in type '{ a: number; }
```

**Comment by mikearnaldi** [comment]
Thanks for opening this, I was aware of it but regarded as "impossible" I did another try and got it working! #113

**Comment by mikearnaldi** [comment]
Released in `@matechs/effect@0.4.21`

**Comment by mohaalak** [comment]
Thank you for all of your works, I love @matechs/effect
